src/shogun/base/SGObject.h

```diff
@@ -1041,6 +1048,124 @@ class CSGObject
 	SGSubscriber* m_subscriber_params;
 };
 
+	class ToStringVisitor : public AnyVisitor
+	{
+	public:
+		ToStringVisitor(std::stringstream* ss) : AnyVisitor(), m_stream(ss)
+		{
+		}
+
+		virtual void on(bool* v)
+		{
+			stream() << (*v ? "true" : "false");
+		}
+		virtual void on(int32_t* v)
+		{
+			stream() << *v;
+		}
+		virtual void on(int64_t* v)
+		{
+			stream() << *v;
+		}
+		virtual void on(float* v)
+		{
+			stream() << *v;
+		}
+		virtual void on(double* v)
+		{
+			stream() << *v;
+		}
+		virtual void on(long double* v)
+		{
+			stream() << *v;
+		}
+		virtual void on(CSGObject** v)
+		{
+			if (*v)
+			{
+				stream() << (*v)->get_name() << "(...)";
+			}
+			else
+			{
+				stream() << "null";
+			}
+		}
+		virtual void on(SGVector<int>* v)
+		{
+			to_string(v);
+		}
+		virtual void on(SGVector<float>* v)
+		{
+			to_string(v);
+		}
+		virtual void on(SGVector<double>* v)
+		{
+			to_string(v);
+		}
+		virtual void on(SGMatrix<int>* mat)
+		{
+			to_string(mat);
+		}
```

Comment:
maybe also move this whole blob to another PR?

Reply:
this has to be reset, with viktors PR I can now get the string from the vector